tests/test-cast-to-overflow.c

```diff
@@ -39,55 +39,234 @@ typedef struct
   int line;
   int round_mode;
   _Decimal128 set;
+  long double expect;
+} d128_type;
+
+typedef struct
+{
+  int line;
+  int round_mode;
+  _Decimal64 set;
   double expect;
-} d_type;
+} d64_type;
+
+typedef struct
+{
+  int line;
+  int round_mode;
+  _Decimal32 set;
+  float expect;
+} d32_type;
+
+/* Decimal 128 to Binary Double 128.  */
+static const d128_type d128to128[] = {
+ {__LINE__, FE_TONEAREST, 1e6000DL, INFINITY},
+ {__LINE__, FE_TONEAREST, -1e6000DL, -INFINITY},
+ {__LINE__, FE_TONEAREST, 1e4933DL, INFINITY},
+ {__LINE__, FE_TONEAREST, -1e4933DL, -INFINITY},
+#if __LDBL_MANT_DIG__ == 106
+ {__LINE__, FE_TONEAREST, 1.797693134862316e308DL, INFINITY},
+ {__LINE__, FE_TONEAREST, -1.797693134862316e308DL, -INFINITY},
+#else
+ {__LINE__, FE_TONEAREST, 1.189731495357232e4932DL, INFINITY},
+ {__LINE__, FE_TONEAREST, -1.189731495357232e4932DL, -INFINITY},
+#endif
+ {__LINE__, FE_TOWARDZERO, 1e6000DL, LDBL_MAX},
+ {__LINE__, FE_TOWARDZERO, -1e6000DL, -LDBL_MAX},
+ {__LINE__, FE_TOWARDZERO, 1e4933DL, LDBL_MAX},
+ {__LINE__, FE_TOWARDZERO, -1e4933DL, -LDBL_MAX},
+#if __LDBL_MANT_DIG__ == 106
+ {__LINE__, FE_TOWARDZERO, 1.797693134862316e308DL, LDBL_MAX},
+ {__LINE__, FE_TOWARDZERO, -1.797693134862316e308DL, -LDBL_MAX},
```

Comment:
This will fail on POWER, with the following message:

```
7   Error:   Expected: "0x1.fffffffffffff7ffffffffffff8p+1023"
              Result:   "0x1.fffffffffffffp+1023"
in: ./tests/test-cast-to-overflow.c:74
```
(similar problems will occur on different lines)

But that's probably due to the fact that _Decimal128 to long double conversions haven't been patched for subnormals (similarly to what was done in commit ID 4f67ffa3c65fed967ad23b8d2aea9921f0bd1cc0, but now for function `trunctdtf`).  Anyhow, that's not your problem and it shouldn't block this pull request, in my opinion.

Reply:
Yes there are some test-fails on power due to the current implementation (also in commit ID 4f67ffa).
Currently trunctdtf.c includes extendddtf.c. I think this is not correct?
Instead there has to be an implementation like trunctddf.c as you've already mentioned.
In issue #71 racardoso said, that he will work on those issues.

But I have anoter question:
Is the __LDBL_MAX__ value correct?
And if yes, why?
See the attached test:
ldbl_tst =     0x1.ffffffffffffffffffffffffff800000p+1023
is greater than
__LDBL_MAX__ = 0x1.fffffffffffff7ffffffffffff800000p+1023

[tst-ldbl-max.c.tar.gz](https://github.com/libdfp/libdfp/files/2206385/tst-ldbl-max.c.tar.gz)

Reply:
@gftg85 @stliibm long double is not working at all https://github.com/libdfp/libdfp/issues/60 also there's other reported issues related to long double such as unimplemented functions.

Reply:
Yes, there are bugs with the long double conversion functions __in libdfp__.  While I understand that this needs to be fixed, it shouldn't block this pull request.  That's what I meant with my comment.

As for Stefan's question about `ldbl_tst` ability to be greater than LDBL_MAX, there's an explanation to it in this bugzilla issue ([PR 70117](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=70117)).  The explanation to the problem is interspersed with another discussion related to the `isinf` function, but comment [#5](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=70117#c5) explains that GCC uses a 106-bit mantissa, and comment [#7](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=70117#c7) suggests that this is probably not going to change.  `ldbl_tst` has a 107-bit mantissa and will compare greater than LDBL_MAX.